File: users-api/src/main/java/net/plshark/errors/DuplicateException.java
package net.plshark.errors;

/**
 * Exception indicating an object with the requested properties already exists
 */
public class DuplicateException extends RuntimeException {

    /**
     * Create a new instance
     * @param message the detail message
     */
    public DuplicateException(String message) {
        super(message);
    }

    /**
     * Create a new instance
     * @param message the detail message
     * @param cause the cause
     */
    public DuplicateException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: user-error-api/src/main/kotlin/net/plshark/usererror/error/DuplicateException.kt
package net.plshark.usererror.error

/**
 * Exception indicating an object with the requested properties already exists
 */
class DuplicateException : UserErrorException {

    constructor() : super()

    constructor(message: String) : super(message)

    constructor(cause: Throwable) : super(cause)

    constructor(message: String, cause: Throwable) : super(message, cause)
}
